cpp/src/Ice/SSL/SchannelTransceiverI.cpp

```diff
@@ -115,29 +131,37 @@ Schannel::TransceiverI::sslHandshake()
     SECURITY_STATUS err = SEC_E_OK;
     if (_state == StateHandshakeNotStarted)
     {
-        SCHANNEL_CRED cred = _engine->newCredentialsHandle(_incoming);
-        if (_localCertificateSelectionCallback)
+        _credentials = _localCredentialsSelectionCallback(_incoming ? _adapterName : _host);
+        if (_rootStore && !_credentials.hRootStore)
+        {
+            _credentials.hRootStore = CertDuplicateStore(_rootStore);
+        }
+
+        if (_credentials.cCreds > 0)
         {
-            _certificate = _localCertificateSelectionCallback(_incoming ? _adapterName : _host);
-            if (_certificate)
+            for (DWORD i = 0; i < _credentials.cCreds; ++i)
             {
-                cred.cCreds = 1;
-                cred.paCred = &_certificate;
+                if (_credentials.paCred[i])
+                {
+                    _allCerts.push_back(CertDuplicateCertificateContext(_credentials.paCred[i]));
+                }
             }
+            _credentials.paCred = &_allCerts[0];
         }
-        cred.hRootStore = _rootStore;
 
-        memset(&_credentials, 0, sizeof(_credentials));
+        _credentials.hRootStore = _rootStore;
+
+        memset(&_credentialsHandle, 0, sizeof(_credentialsHandle));
```

Comment:
I think I can remove this memset, we already have `_credentialsHandle{0, 0}` in the ctor.